cmd/pbstatuser/main.go

```diff
@@ -20,25 +26,179 @@ import (
 	"github.com/rs/zerolog/log"
 )
 
+// This needs to be done in 4 goroutines.
+// 1) Reads messages, calls sources to get auth type and ARN and then enqueue the work
+// in one of three channels (AWS, Azure, GCP)
+// 2) Reads from AWS channel, performs availability check, has configurable throttling.
+// 3) Reads from Azure channel, performs availability check, has configurable throttling.
+// 4) Reads from GCP channel, performs availability check, has configurable throttling.
+
+// Beware there is no database connection, no job queue, most of what api/worker processes have
+// is not available because it should not be needed. The worker will simply use SDKs to check
+// statuses and send out results via kafka.Send function to Sources.
+// Also make sure that all goroutines are closed correctly on context cancel. A WaitGroup
+// would be a really good here: https://gist.github.com/fracasula/b579d52daf15426e58aa133d0340ccb0
+
+const ChannelBuffer = 1024
```

Comment:
I suggest to either bring this really down to like 32 or 16 because if we slow things down, any larger queue could actually make shutting down operation to take a lot of time. Imagine if we slow down to operations per second, then to finish up 1k buffer it would take 200 seconds while 32 would be just 6 seconds. I think k8s will not allow too much time for shutdown.

Even unbuffered channel is a good option here.

Reply:
I don't understand why do you think an unbuffered channel is better

Reply:
Because the processing will likely be delayed artificially therefore the receiver/sources requester goroutine will likely be always faster, thus there will never be lack of messages in the channel. The checker goroutines will not have to wait too much. As I said, small buffer will do, but 32 is probably the highest number I would go for.